gtest/src/manager.rs

```diff
@@ -948,6 +948,40 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    fn remove_reservation(&mut self, id: ProgramId, reservation: ReservationId) -> Option<()> {
+        let res = self.adjust_genuine_program(id, |genuine_program| {
+            let genuine_program = genuine_program.expect("no genuine program was found");
+            genuine_program
+                .gas_reservation_map
+                .remove(&reservation)
+                .map(|_| ())
+                .or_else(|| {
+                    log::error!("Tried to remove non-existent reservation");
+
+                    None
+                })
+        });
+
+        res.map(|_| {
+            self.gas_tree
+                .consume(reservation)
+                .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+        })
+    }
+
+    pub(crate) fn adjust_genuine_program<R, F: FnOnce(Option<&mut GenuineProgram>) -> R>(
```

Comment:
```suggestion
    pub(crate) fn update_genuine_program<R, F: FnOnce(Option<&mut GenuineProgram>) -> R>(
```
for consistency with program storage

Reply:
done